### PR TITLE
[#572] Update Infrastructure Mappings list to display OSP mappings

### LINKED ISSUE #572: [OSP] Update Overview Infra Mapping List
The infra mapping ListView needs to be updated to be compatible with OSP mappings

### PR DESCRIPTION
**Pull request by mturley**
Closes #572

### CODE REVIEW

File: app/javascript/react/screens/App/Overview/components/InfrastructureMappingsList/helpers.js [hunk 1]
```diff
@@ -47,15 +47,19 @@ export const mapInfrastructureMappings = (transformation_mapping_items, clusters
...
-  const networkMappingItems = transformation_mapping_items.filter(
-    item => item.destination_type.toLowerCase() === 'lan'
-  );
+  const clusterMappingItems = transformation_mapping_items.filter(item => {
+    const destination = item.destination_type.toLowerCase();
+    return destination === 'emscluster' || destination === 'cloudtenant';
+  });
+  const datastoreMappingItems = transformation_mapping_items.filter(item => {
+    const destination = item.destination_type.toLowerCase();
+    return destination === 'storage' || destination === 'cloudvolume';
```
Comment (michaelkro) [inline app/javascript/react/screens/App/Overview/components/InfrastructureMappingsList/helpers.js]: Now that https://github.com/ManageIQ/manageiq-v2v/pull/673 is in, might as well update this now

Comment (mturley) [reply app/javascript/react/screens/App/Overview/components/InfrastructureMappingsList/helpers.js]: I'm on it

**Comment by mturley** [comment]
@michaelkro , @vconzola said he will mock up some label changes for this too, so i'm waiting on that before we merge this.

**Comment by mturley** [comment]
Ahh! Good catch... @michaelkro it looks like it's displaying the RHV version of all three (networks, datastores, clusters). This will be a tricky fix, I need to fetch `cloud_tenant`, `cloud_volume_type` and `cloud_network`, and possibly refactor the mapping code a bit.

**Comment by mturley** [comment]
@vconzola, when you return on Monday I just want to run this by you. In addition to the labels you mocked up in #572, because this data was being displayed and it was easier than adding a bunch of conditions specific to OSP just to hide things, I added `Provider` to the target labels for datastores and networks, and I added `Cluster` to the source labels for datastores and networks. Let me know if these are okay:

screenshot 2018-10-05 16 32 02

screenshot 2018-10-05 16 31 58

screenshot 2018-10-05 16 31 52

**Comment by vconzola** [comment]
@mturley This looks good. We'll need to create a separate issue to add the source provider type to the list view too.

**Comment by mturley** [comment]
@vconzola I think we already have one: https://github.com/ManageIQ/manageiq-v2v/issues/681

**Comment by vconzola** [comment]
@mturley Thanks. I forgot I'd already opened one.

File: app/javascript/packs/polyfills.js [hunk 2]
```diff
@@ -4,3 +4,4 @@
 import 'core-js/es6/promise';
 import 'core-js/es6/set';
 import 'core-js/es6/map';
+import 'core-js/fn/array/flat-map';
```
Comment (michaelkro) [inline app/javascript/packs/polyfills.js]: I think this can be removed, no?

Comment (mturley) [reply app/javascript/packs/polyfills.js]: Ah, yep. Good catch.

**Comment by mturley** [comment]
Big thanks to @michaelkro for the help with cedbe4c ! That helper function is a beast. We'll need to revisit it.

**Pull request merged**